core/src/main/scala/com/github/gpoirier/barrage/GameState.scala

```diff
@@ -54,25 +55,19 @@ object GameState {
   private def forActivePlayerF(f: PlayerState => PlayerState): StateT[F, GameState, Unit] =
     StateT.modify[F, GameState](forActivePlayer(f))
 
-  private[barrage] def resolveCommand: Command => StateT[F, GameState, Unit] = {
-    case Command(action: Action.Workshop, Nil) =>
-      lens.workshop composeWith Workshop.forAction(action) flatMap { column =>
-        lens.currentPlayerState composeWith {
-          for {
-            _ <- PlayerState.payCost(action.cost(column))
-            _ <- PlayerState.spin(action.spin)
-          } yield ()
-        }
+  private[barrage] def resolveCommand(command: Command): StateT[F, GameState, Unit] = {
+    for {
+      cost <- command.action.take
+      _ <- lens.currentPlayerState composeWith PlayerState.payCost(cost)
+      _ <- command.action.rewards.value.traverse {
+        case reward: CompleteReward =>
+          StateT.modify[Result, GameState](reward.update)
+        case reward: IncompleteReward =>
+          // FIXME
+          val selector = command.rewardsHandlers.head
+          reward.update(selector)
```

Comment:
This is a very simplistic implementation that expect only one reward selector, and only if the reward supports it. It needs to be extended to support multiple selectors and validate that there's no extra selectors.